front/app/containers/Admin/pagesAndMenu/containers/GenericHeroBannerForm/BannerImageFields/OverlayControls.tsx

```diff
@@ -0,0 +1,151 @@
+const StyledBox = styled(Box)`
+  position: relative;
+
+  ::before,
+  ::after {
+    content: '';
+    display: block;
+    position: absolute;
+    width: 0;
+    height: 0;
+    border-style: solid;
+  }
+
+  ::after {
+    top: -20px;
+    left: 25px;
+    border-color: transparent transparent #fff transparent;
+    border-width: 10px;
+  }
+
+  ::before {
+    top: -22px;
+    left: 24px;
+    border-color: transparent transparent ${colors.grey300} transparent;
+    border-width: 11px;
+  }
+`;
+
+interface Props {
+  bannerOverlayOpacity: BannerImageFieldsProps['bannerOverlayOpacity'];
+  bannerOverlayColor: BannerImageFieldsProps['bannerOverlayColor'];
+  onOverlayOpacityChange: BannerImageFieldsProps['onOverlayOpacityChange'];
+  onOverlayColorChange: BannerImageFieldsProps['onOverlayColorChange'];
+}
+
+const OverlayControls = ({
+  bannerOverlayOpacity,
+  bannerOverlayColor,
+  onOverlayOpacityChange,
+  onOverlayColorChange,
+}: Props) => {
+  const [overlayEnabled, setOverlayEnabled] = useState(
+    typeof bannerOverlayOpacity === 'number'
+  );
+  const theme = useTheme();
+  const { formatMessage } = useIntl();
+
+  const handleOverlayEnabling = () => {
+    // Ideally we should also change the color here
+    // but if we call with two individual setting handlers,
+    // they will overwrite each other setting one of the two
+    // values to null. We need a new handler that updates
+    // multiple settings at the same time but this requires
+    // refactoring of components uses in the GenericHeroBannerForm
+    // to home/custom page specific versions to make the types fit.
+    if (overlayEnabled) {
+      handleOverlayOpacityOnChange(null);
+    } else {
+      handleOverlayOpacityOnChange(
+        bannerOverlayOpacity || theme.signedOutHeaderOverlayOpacity
+      );
+    }
+
+    setOverlayEnabled((overlayEnabled) => !overlayEnabled);
+  };
+
+  const handleOverlayOpacityOnChange = (
+    opacity: Props['bannerOverlayOpacity']
+  ) => {
+    onOverlayOpacityChange(opacity);
+  };
+
+  const debounceHandleOverlayOpacityOnChange = debounce(
+    handleOverlayOpacityOnChange,
+    1
+  );
+
+  const debouncedHandleOverlayOpacityOnChange = useMemo(
+    () => debounceHandleOverlayOpacityOnChange,
+    [debounceHandleOverlayOpacityOnChange]
+  );
+
+  return (
+    <>
+      <Box mb={overlayEnabled ? '20px' : '0'}>
+        <Toggle
+          onChange={handleOverlayEnabling}
+          checked={overlayEnabled}
+          label={
+            <Box color={colors.blue500}>
+              {formatMessage(messages.overlayToggleLabel)}
+            </Box>
+          }
+        />
+      </Box>
+      {/*
+        We check for typeof of opacity because 0 would coerce to false.
+        We don't do a similar thing for the color because we don't set
+        it in the toggle hander, so we handle the type check with the
+        theme default as fallback
+      */}
+      {overlayEnabled && typeof bannerOverlayOpacity === 'number' && (
```

Comment:
`typeof` is a little bit ambiguous (though a comment handles it). Maybe this would be more explicit

```suggestion
      {overlayEnabled && bannerOverlayOpacity !== null && (
```

Reply:
What if we change `bannerOverlayOpacity`'s type to include `undefined`? The `!== null` check will gladly let it pass.

Reply:
We explicitely want it to be a number and nothing else.

Reply:
> What if we change bannerOverlayOpacity's type to include undefined?

TS will complain because `RangeInput` accepts only numbers. But up to you.

Reply:
🙂 

> Edit: the component prop types would complain in that case though. But I like to be explicit.